### PR TITLE
[16.0][ADD] stock_customer_deposit_elaboration

### PR DESCRIPTION
**Pull request by EmilioPascual**
This module is a glue module and it was developed because the modules **sale_elaboration** and **stock_customer_deposit** create a compute method for the order line *route_id* field.

As odoo does not have a compute core method for this field, it cannot be inherited so if you install both modules, one method will be overwritten and one behaviour will be lost.

First it is necessary to merge this pr https://github.com/OCA/stock-logistics-workflow/pull/1503

@yajo @Shide please review

MT-5513 @moduon

### CODE REVIEW

File: stock_customer_deposit_elaboration/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "summary": "Glue module betwen stock_customer_deposit and sale_elaboration",
+    "version": "16.0.1.0.0",
+    "development_status": "Alpha",
+    "category": "Inventory/Delivery",
+    "website": "https://github.com/OCA/stock-logistics-workflow",
+    "author": "Moduon, Odoo Community Association (OCA)",
+    "maintainers": ["EmilioPascual", "rafaelbn"],
+    "license": "LGPL-3",
+    "application": False,
+    "installable": True,
```
Comment (yajo) [inline stock_customer_deposit_elaboration/__manifest__.py]: issue: glue modules should auto-install

```suggestion
    "auto_install": True,
```

File: stock_customer_deposit_elaboration/models/sale_order_line.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        self.ensure_one()
+        if (
+            self.warehouse_id.use_customer_deposits
+            and self.product_id.type == "product"
+            and self.order_id.customer_deposit
+        ):
+            return self.warehouse_id.customer_deposit_route_id
+        elif self.elaboration_ids.route_ids[:1]:
+            return self.elaboration_ids.route_ids[:1]
+        return
```
Comment (yajo) [inline stock_customer_deposit_elaboration/models/sale_order_line.py]: suggestion: this is mostly the same

```suggestion
        return self.elaboration_ids.route_ids[:1]
```

**Review by yajo — CHANGES_REQUESTED** [review]

**Review by yajo — APPROVED** [review]
Code review.

**Comment by fcvalgar** [comment]
@EmilioPascual you can check to have a runboat available. Thank you.

**Comment by EmilioPascual** [comment]
> @EmilioPascual you can check to have a runboat available. Thank you.

Sorry @fcvalgar, but until this PR https://github.com/OCA/stock-logistics-workflow/pull/1503 is merged, runboat will not work.

**Comment by rafaelbn** [comment]
@EmilioPascual https://github.com/OCA/stock-logistics-workflow/pull/1503 is merging please rebase when finished! Thank you!

**Review by rafaelbn — APPROVED** [review]
/ocabot merge patch

**Comment by rafaelbn** [comment]
Please @EmilioPascual @yajo  review https://github.com/OCA/stock-logistics-workflow/commits/7ad2506e581f6ca4a3fe386a7f0836559af788aa

**Comment by yajo** [comment]
There are incompatible modules, please @EmilioPascual declare them rebels.

**Review by fcvalgar — APPROVED** [review]
Functional review.

LGTM, thank you @EmilioPascual

**Review by Shide — APPROVED** [review]

**Comment by EmilioPascual** [comment]
> There are incompatible modules, please @EmilioPascual declare them rebels.

Done!. Thank you  @yajo

**Pull request merged**